mesonbuild/compilers.py

```diff
@@ -801,10 +779,40 @@ def get_soname_args(self, prefix, shlib_name, suffix, path, soversion, is_shared
     def split_shlib_to_parts(self, fname):
         return None, fname
 
-    # The default behavior is this, override in
-    # OSX and MSVC.
+    def build_unix_rpath_args(build_dir, from_dir, rpath_paths, install_rpath):
+            if not rpath_paths and not install_rpath:
+                return []
+            # The rpaths we write must be relative, because otherwise
+            # they have different length depending on the build
+            # directory. This breaks reproducible builds.
+            rel_rpaths = []
+            for p in rpath_paths:
+                if p == from_dir:
+                    relative = '' # relpath errors out in this case
+                else:
+                    relative = os.path.relpath(p, from_dir)
+                rel_rpaths.append(relative)
+            paths = ':'.join([os.path.join('$ORIGIN', p) for p in rel_rpaths])
+            if len(paths) < len(install_rpath):
+                padding = 'X' * (len(install_rpath) - len(paths))
+                if not paths:
+                    paths = padding
+                else:
+                    paths = paths + ':' + padding
+            args = ['-Wl,-rpath,' + paths]
+            if get_compiler_is_linuxlike(self):
+                # Rpaths to use while linking must be absolute. These are not
+                # written to the binary. Needed only with GNU ld:
+                # https://sourceware.org/bugzilla/show_bug.cgi?id=16936
+                # Not needed on Windows or other platforms that don't use RPATH
+                # https://github.com/mesonbuild/meson/issues/1897
+                lpaths = ':'.join([os.path.join(build_dir, p) for p in rpath_paths])
```

Comment:
__[flake8]__
* [F841] local variable 'lpaths' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->